src/main/kotlin/org/rust/ide/console/RsConsoleCodeFragmentContext.kt

```diff
@@ -10,42 +10,61 @@ import com.intellij.openapi.project.DumbService
 import com.intellij.openapi.project.Project
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.lang.core.psi.*
-import org.rust.lang.core.psi.ext.descendantOfTypeStrict
+import org.rust.lang.core.psi.ext.*
 import org.rust.openapiext.toPsiFile
 
 class RsConsoleCodeFragmentContext {
 
-    private val topLevelElements: MutableMap<String, String> = mutableMapOf()
-    private val allCommandsText: StringBuilder = StringBuilder()
+    private val itemsNames: Set<String> = mutableSetOf()
```

Comment:
Looks like `itemsNames` is always empty, so `commands` contains only one element and nested scopes are not used later

Reply:
Thanks! Should be fixed now. Also I updated tests, so now they behave closer to real REPL